algorithmic_efficiency/workloads/librispeech_deepspeech/librispeech_jax/models.py

```diff
@@ -0,0 +1,730 @@
+      concatenate the outputs from the two directions.
+    cell_type: The LSTM cell class to use. Default:
+      `flax.linen.OptimizedLSTMCell`. If you use hidden_size of >2048, consider
+      using `flax.linen.LSTMCell` instead, since the optimized LSTM cell works
+      best for hidden sizes up to 2048.
+    cell_kwargs: Optional keyword arguments to instantiate the cell with.
+  """
+  hidden_size: int
+  num_layers: int = 1
+  dropout_rate: float = 0.
+  recurrent_dropout_rate: float = 0.
+  bidirectional: bool = False
+  cell_type: Any = nn.OptimizedLSTMCell
+  cell_kwargs: Mapping[str, Any] = flax.core.FrozenDict()
+
+  @nn.compact
+  def __call__(
+      self,
+      inputs: Array,
+      lengths: Array,
+      initial_states: Optional[Sequence[StateType]] = None,
+      deterministic: bool = False) -> Tuple[Array, Sequence[StateType]]:
+    """Processes an input sequence with an LSTM cell.
+
+    Example usage:
+    ```
+      inputs = np.random.normal(size=(2, 3, 4))
+      lengths = np.array([1, 3])
+      outputs, final_states = LSTM(hidden_size=10).apply(rngs, inputs, lengths)
+    ```
+
+    Args:
+      inputs: The input sequence <float32>[batch_size, sequence_length, ...]
+      lengths: The lengths of each sequence in the batch. <int64>[batch_size]
+      initial_states: The initial states for the cells. You must provide
+        `num_layers` initial states (when using bidirectional, `num_layers *
+        2`). These must be ordered in the following way: (layer_0_forward,
+          layer_0_backward, layer_1_forward, layer_1_backward, ...). If None,
+          all initial states will be initialized with zeros.
+      deterministic: Disables dropout between layers when set to True.
+
+    Returns:
+      The sequence of all outputs for the final layer, and a list of final
+      states (h, c) for each cell and direction, ordered first by layer number
+      and then by direction (first forward, then backward, if bidirectional).
+    """
+    return GenericRNN(
+        cell_type=self.cell_type,
+        hidden_size=self.hidden_size,
+        num_layers=self.num_layers,
+        dropout_rate=self.dropout_rate,
+        recurrent_dropout_rate=self.recurrent_dropout_rate,
+        bidirectional=self.bidirectional,
+        cell_kwargs=self.cell_kwargs,
+        name='LSTM')(
+            inputs,
+            lengths,
+            initial_states=initial_states,
+            deterministic=deterministic)
+
+
+class BatchRNN(nn.Module):
+  """Implements a single deepspeech encoder layer.
+  """
+  config: DeepspeechConfig
+
+  @nn.compact
+  def __call__(self, inputs, input_paddings, train):
+    config = self.config
+
+    inputs = BatchNorm(config.encoder_dim,
+                       config.dtype,
+                       config.batch_norm_momentum,
+                       config.batch_norm_epsilon)(inputs, input_paddings, train)
+    lengths = jnp.sum(1 - input_paddings, axis=-1, dtype=jnp.int32)
+
+    output, _ = LSTM(
```

Comment:
return just outputs and delete the `, _`

Reply:
deleted